add_keys.py

```diff
@@ -0,0 +1,204 @@
+
+ARTIFACTS_DIR = './assets'
+POOL_ADDRESS = "0xae7ab96520de3a18e5e111b5eaab095312d7fe84"
+POOL_ARTIFACT_FILE = 'Lido.json'
+REGISTRY_ARTIFACT_FILE = 'NodeOperatorsRegistry.json'
+GAS_LIMIT=13500000
+
+
+def init_provider(eth1_provider):
+    '''
+    Initialise the Web3 provider.
+    '''
+    if eth1_provider.startswith('http'):
+        provider = HTTPProvider(eth1_provider)
+    elif eth1_provider.startswith('ws'):
+        provider = WebsocketProvider(eth1_provider)
+    else:
+        logging.error('Unsupported ETH provider!')
+        sys.exit(1)
+
+    return Web3(provider)
+
+
+def init_contracts(w3: Web3, pool_address):
+    '''
+    Initialise contracts.
+    '''
+
+    if not Web3.isChecksumAddress(pool_address):
+        pool_address = Web3.toChecksumAddress(pool_address)
+
+    pool_abi_path = os.path.join(ARTIFACTS_DIR, POOL_ARTIFACT_FILE)
+    registry_abi_path = os.path.join(ARTIFACTS_DIR, REGISTRY_ARTIFACT_FILE)
+
+    # Get Pool contract
+    with open(pool_abi_path, 'r') as file:
+        abi = json.load(file)
+        file.close()
+    pool = w3.eth.contract(abi=abi['abi'], address=pool_address)  # contract object
+
+    # Get Registry contract
+    registry_address = pool.functions.getOperators().call()
+    logging.info(f'{registry_address=}')
+
+    with open(registry_abi_path, 'r') as file:
+        abi = json.load(file)
+        file.close()
+    return w3.eth.contract(abi=abi['abi'], address=registry_address)
+
+
+def init_account(w3: Web3):
+    '''
+    Initialise the signer.
+    '''
+    privkey = os.getenv('PRIV_KEY')
+    account = w3.eth.account.from_key(privkey)
+    logging.info(f'Account: {account.address}')
+    return account
+
+
+
+@click.command()
+@click.option("--nonce", default=-1, type=int, help="Nonce to use (default uses next nonce on chain)")
+@click.option("--gas", default=GAS_LIMIT, type=int, help="Gas limit - defaults to {}, which is suitable for submitting 100 keys.".format(GAS_LIMIT))
+@click.option("--operator", help="Operator ID.", type=int)
+@click.option("--eth1-uri", help="Ethereum node address", type=str)
+@click.option("--pool-address", help="Lido pool contract address - defaults to mainnet contract address", type=str, default=POOL_ADDRESS)
+@click.argument('filename')
+def main(filename, nonce, operator, gas, eth1_uri, pool_address):
+    '''
+    Main entrypoint.
+    '''
+    try:
+        if eth1_uri == None:
+            eth1_uri = os.environ.get('WEB3_PROVIDER_URI', None)
+            if eth1_uri == None:
+                raise Exception("Must pass --eth1-uri flag or set WEB3_PROVIDER_URI environment variable")
+        w3 = init_provider(eth1_uri)
+        registry = init_contracts(w3, pool_address)
+        account = init_account(w3)
+
+        if operator is None:
+            raise Exception("--operator flag is required")
+
+        with open(filename, 'r') as f:
+            keyfile = json.load(f)
+            f.close()
+
+        keys = [x.get('pubkey') for x in keyfile]
```

Comment:
This can result in `None`s in `key`; `.get` returns `None` when the key is not present, and I suspect that would break things down the line if it happened. Use `x['pubkey']` to make it raise a `KeyError` here instead.